File: apps/web/lib/types/resend-readiness.ts
export type ResendReadinessFreshness = "fresh" | "stale" | "never_checked"
export type ResendReadinessProbeStatus = "succeeded" | "limited" | "failed"
export type ResendReadinessCapabilityStatus =
    | "ready"
    | "needs_attention"
    | "limited"
    | "unknown"
    | "not_configured"
export type ResendReadinessCheckStatus = "idle" | "queued" | "running"
export type ResendReadinessIssueCode =
    | "admission_unavailable"
    | "credential_rejected"
    | "credential_unavailable"
    | "delivery_events_missing"
    | "domain_not_verified"
    | "engagement_events_missing"
    | "invalid_provider_response"
    | "limited_visibility"
    | "provider_unavailable"
    | "sending_disabled"
    | "snapshot_stale"
    | "timeout"
    | "webhook_disabled"
    | "webhook_missing"

export interface ResendReadinessSnapshot {
    freshness: ResendReadinessFreshness
    probe_status: ResendReadinessProbeStatus | null
    overall_status: ResendReadinessCapabilityStatus
    domain_status: ResendReadinessCapabilityStatus
    webhook_status: ResendReadinessCapabilityStatus
    sending_status: ResendReadinessCapabilityStatus
    delivery_tracking_status: ResendReadinessCapabilityStatus
    engagement_tracking_status: ResendReadinessCapabilityStatus
    verified_domain_count: number
    enabled_webhook_count: number
    issue_codes: ResendReadinessIssueCode[]
    checked_at: string | null
    last_success_at: string | null
}

export interface ResendReadinessEnvelope {
    check_status: ResendReadinessCheckStatus
    last_snapshot: ResendReadinessSnapshot
}

File: apps/api/app/schemas/resend_readiness.py
"""Sanitized public contracts for cached Resend readiness checks."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


ReadinessCheckStatus = Literal["idle", "queued", "running"]
ReadinessFreshness = Literal["fresh", "stale", "never_checked"]
ReadinessProbeStatus = Literal["succeeded", "limited", "failed"]
ReadinessCapabilityStatus = Literal[
    "ready",
    "needs_attention",
    "limited",
    "unknown",
    "not_configured",
]


class ResendReadinessSnapshotResponse(BaseModel):
    """Provider-safe cached result; no routing, credential, or provider identifiers."""

    model_config = ConfigDict(from_attributes=True)

    freshness: ReadinessFreshness
    probe_status: ReadinessProbeStatus | None
    overall_status: ReadinessCapabilityStatus
    domain_status: ReadinessCapabilityStatus
    webhook_status: ReadinessCapabilityStatus
    sending_status: ReadinessCapabilityStatus
    delivery_tracking_status: ReadinessCapabilityStatus
    engagement_tracking_status: ReadinessCapabilityStatus
    verified_domain_count: int
    enabled_webhook_count: int
    issue_codes: tuple[str, ...]
    checked_at: datetime | None
    last_success_at: datetime | None


class ResendReadinessEnvelope(BaseModel):
    """Current durable-check state plus the latest still-visible cached result."""

    model_config = ConfigDict(from_attributes=True)

    check_status: ReadinessCheckStatus
    last_snapshot: ResendReadinessSnapshotResponse
